sympy/ntheory/digits.py

```diff
@@ -0,0 +1,107 @@
+from __future__ import print_function, division
+
+import math
+import re
+
+from collections import (
+    Counter,
+    OrderedDict,
+)
+
+
+def count_digits(n, base=10):
+    """
+    Returs an ordered dict. frequency counter for the digits of a decimal
+    integer ``n`` representing an integer in a base ``base``, where the
+    base digits are mapped to the positive decimal integers, e.g. the
+    base ``11`` integer ``10321``, with digits ``10``, ``3``, ``2``, ``1``
+    (from most significant to least), is equal to the decimal integer
+    ``13696``, as ``10 * 11**3 + 3 * 11**2 + 2 * 11**1 + 1 * 11**0 = 13696``.
+
+    For binary, octal or hexadecimal bases the integer ``n``
+    can also be specified as a numeric literal in those bases, e.g. ``0xF321``.
+
+    Examples
+    ========
+
+    >>> from sympy.ntheory import count_digits
+
+    >>> count_digits(1903413094, 10)
+    {0: 2, 1: 2, 3: 2, 4: 2, 9: 2}
+
+    >>> count_digits(122333, 10)
+    {1: 1, 2: 2, 3: 3}
+
+    >>> count_digits(0b1001001, 2)
+    {0: 4, 1: 3}
+
+    >>> count_digits(0xF321, 16)
+    {1: 1, 2: 1, 3: 1, 15: 1}
+
+    The base 11 number 10321, with digits 10, 3, 2, 1 (from most significant digit)
+    >>> count_digits(13696, 11)
+    {1: 1, 2: 1, 3: 1, 10: 1}
+
+    The base 100 number 990512, with digits 99, 0, 51, 2 (from most significant digit)
+    >>> count_digits(99005102, 100)
+    {0: 1, 2: 1, 51: 1, 99: 1}
+
+    """
+    m = math.floor(math.log(n, base)) + 1
+
+    return OrderedDict(
+        sorted(
+            Counter(((n // base ** k) % base for k in range(m))).items(),
+            key=lambda t: t[0]
+        )
+    )
+
+
+def is_palindromic_integer(n, base=10):
```

Comment:
IMO, if you import from `digits` you shouldn't have to say 'integer' when calling the function. But the function can check its input.

Reply:
I had pushed on fully qualified names, because I see lots of things are eventually getting imported with the sympy global namespace, cluttering the place where other names can jump in.

If the things are not made to be imported under the global namespace, but make use of the module namespace, I’d more agree on making the names shorter.

Reply:
To avoid name collision imports can of course be aliased.